vmm/src/api/http/mod.rs

```diff
@@ -314,6 +317,188 @@ fn handle_http_request(
+                            .map_err(|e| {
+                                error!("Error applying landlock to http-worker thread: {:?}", e);
+                                exit_evt.write(1).ok();
+                                e
+                            })?;
+                    }
+
+                    std::panic::catch_unwind(AssertUnwindSafe(move || {
+                        loop {
+                            let id = n;
+                            match request_rx.recv() {
+                                Ok(msg) => {
+                                    // Process the server request
+                                    let response = msg.process(|request| {
+                                        handle_http_request(request, &api_notifier, &api_sender)
+                                    });
+
+                                    // Send the response to the HTTP server thread together with this
+                                    // threads id.
+                                    let response = WorkerAnswer { id, response };
+                                    if let Err(e) = response_tx.send(response) {
+                                        error!(
+                                            "HTTP worker thread {id}: error sending response {}",
+                                            e
+                                        );
+                                        break;
+                                    }
+
+                                    // Notify the HTTP server thread.
+                                    response_event.write(1).ok();
+                                }
+                                Err(e) => {
+                                    error!(
+                                        "HTTP worker thread {id}: error receiving request {}",
+                                        e
+                                    );
+                                    break;
+                                }
+                            }
+                        }
+                    }))
+                    .map_err(|_| {
+                        error!("http-worker thread {n} panicked");
+                        exit_evt.write(1).ok()
+                    })
+                    .ok();
+
+                    Ok(())
+                })
+                .map_err(VmmError::HttpThreadSpawn)?;
+
+            threads.push(thread);
+            threads_ready.push(true);
+            request_txs.push(request_tx);
+        }
+
+        Ok(Self {
+            threads,
+            threads_ready,
+            request_txs,
+            request_queue: Vec::new(),
+            next_request: 0,
+            response_rx,
+            response_event,
+        })
+    }
+
+    /// Returns a channel to a worker thread that is ready for more work, if
+    /// any workers are ready.
+    fn next_worker(&mut self) -> std::option::Option<&std::sync::mpsc::SyncSender<ServerRequest>> {
```

Comment:
to me, this looks overly complicated. I actually thought about a solution where all worker threads blockingly `receiver.recv()` on a channel with Http Requests. 

Then we also don't need a round robin algorithm. The thread that is ready picks up the request from the channel. That's a typical pattern for thread pools in Rust.

Is there a reason you decided against this?